.github/workflows/nightly.yml

```diff
@@ -0,0 +1,189 @@
+name: Nightly testing workflows for colcon-bundle
+on:
+  # Build at minute 43 hour 17 (9:43am PST)
+  # Minutes was chosen randomly
+  schedule:
+    - cron: '43 17 * * *'
+
+jobs:
+  tox_unittest:
+    runs-on: ubuntu-18.04
+    env:
+      TOXENV: unittest
+    steps:
+    - name: Checkout sources
+      uses: actions/checkout@v2
+    - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        aws-region: ${{ secrets.AWS_REGION }}
+    - name: Setup Python
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.5
+    - name: Install dependencies
+      run: |
+        sudo apt update && sudo apt install -y enchant
+        pip install -U tox setuptools==44.0.0
+    - name: Run tests
+      run: tox
+    - name: Log Test Failure
+      uses: ros-tooling/action-cloudwatch-metrics@0.0.1
+      with:
+        status: 'failure'
+      if: failure() && github.event_name == 'schedule'
+    - name: Log Test Success
+      uses: ros-tooling/action-cloudwatch-metrics@0.0.1
+      with:
+        status: 'success'
+      if: success() && github.event_name == 'schedule'
+
+  backwards_compatability_test:
+    runs-on: ubuntu-18.04
+    env:
+      TOXENV: py35-pypi
+      ROS_DISTRO: kinetic
+      TEST: backwards
+    steps:
+    - name: Checkout sources
+      uses: actions/checkout@v2
+    - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        aws-region: ${{ secrets.AWS_REGION }}
+    - name: Setup Python
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.5
+    - name: Install dependencies
+      run: |
+        sudo apt update && sudo apt install -y enchant
+        pip install -U tox setuptools==44.0.0
+    - name: Run tests
+      run: tox
+    - name: Log Test Failure
+      uses: ros-tooling/action-cloudwatch-metrics@0.0.1
+      with:
+        status: 'failure'
+      if: failure() && github.event_name == 'schedule'
+    - name: Log Test Success
+      uses: ros-tooling/action-cloudwatch-metrics@0.0.1
+      with:
+        status: 'success'
+      if: success() && github.event_name == 'schedule'
+
+  double_test:
```

Comment:
what is "double" test and how is it not handled  by the matrix in ros1_tests

Reply:
I'm not quite sure what it does; the purpose of this PR was to improve the GitHub workflow by refactoring what we currently have.

I've opened the issue: https://github.com/colcon/colcon-bundle/issues/148 to resolve this

Reply:
all right - i see now that this is a full duplicate of the PR one plus cloudwatch. it feels really fragile to be completely duplicated with one difference, let's move forward as long as you're brainstorming how to deduplicate

Reply:
I think I can skip steps using `if`, opening a PR to test this

Reply:
Opened PR https://github.com/colcon/colcon-bundle/pull/149 which skips AWS configuration instead of duplicating work. I'll close this PR out if that workflow is working and its agreed upon that its a better solution.

Reply:
@zmichaels11 @emersonknapp  See the [`.travis.yml`](https://github.com/colcon/colcon-bundle/blob/master/.travis.yml) to understand what each of these tests are doing. Each one is matrix of the tests using a specific Docker build.